client.go

```diff
@@ -83,3 +93,97 @@ func newClient(ctx context.Context, addr string, opts ...client.ClientOpt) (*cli
 	}))
 	return client.New(ctx, "", opts...)
 }
+
+// dockerDir returns the path to the user's Docker config dir.
+func dockerDir() string {
+	dockerConfigDir := os.Getenv("DOCKER_CONFIG")
+	if dockerConfigDir != "" {
+		return dockerConfigDir
+	}
+
+	home := os.Getenv("HOME")
+	if home == "" {
+		return ""
+	}
+
+	return filepath.Join(home, ".docker")
+}
+
+// dockerConf returns the path to the user's Docker config.json.
+func dockerConf(dockerDir string) string {
+	if dockerDir == "" {
+		return ""
+	}
+	return filepath.Join(dockerDir, "config.json")
+}
+
+// dockerHost returns the value of the DOCKER_HOST env var, or the default.
+func dockerHost() (string, error) {
```

Comment:
Export this, so newt can use it in `newtdocker`?

Reply:
Not sure, the newt versions of this use the ctx.Env struct, but this uses os.Getenv, I suppose I could add an abstraction here for this...

Reply:
I added a `WithEnv` and `LookupEnv`, `GetEnv` functions to allow env usage optionally via context values